### PR TITLE
Update to support list/working dir.

### PR DESCRIPTION
**Pull request by mneme**
1. Changed to json formatter and stole the parsing code for flycheck-eslint.
2. Added :working-directory to use local xo.
3. Changed to stdin-parsing with filename to get local paths to resolve.

### CODE REVIEW

File: flycheck-xo.el [hunk 1]
```diff
@@ -39,33 +39,50 @@
 
 (flycheck-def-args-var flycheck-xo-args xo)
 
-(flycheck-define-checker xo
-  "A checker using xo.
+(defun flycheck-parse-xo (output checker buffer)
```
Comment (alexmurray) [inline flycheck-xo.el]: The more conventional name would be flycheck-xo-parse since is then using the package namespace prefix - can you please change it?

**Comment by alexmurray** [comment]
Thanks - I've added a few comments. Why the change to json - does this add much benefit?

**Comment by mneme** [comment]
json: I didn't get flycheck-error-list to work with --compact view and the old parsing. I might just be an idiot though. :)

File: flycheck-xo.el (hunk 1 above)
Comment (mneme) [reply flycheck-xo.el]: done and done

**Comment by alexmurray** [comment]
Thanks - I haven't used xo in a while so this could have easily bitrotted - cheers.

**Pull request merged**